### PR TITLE
feat: support of custom properties for subscribers #2393

### LINKED ISSUE #2393: 🚀 Feature: support of custom properties for subscribers
### 🔖 Feature description

Right now, we have fixed schema while creating a new subscriber. In few use cases, we have to store few more information in subscribers meta data. But it is not possible.

### 🎤 Why is this feature needed ?

To store custom properties in subscriber

### ✌️ How do you aim to achieve this?

This [API](https://docs.novu.co/api/create-subscriber/) endpoint expects a fixed schema as body, We can include, custom filed of JSON type here to store subscriber custom properties.

### 🔄️ Additional Information

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by tsssdev**
### What change does this PR introduce?

Applied changes to save custom properties with subscriber, right now added this logic only for direct subscriber create and update paths only. While triggering also we are creating/updating subscribers but not added this custom properties logic in that path as I felt it is better not to overload trigger function with additional payload, is this OK or we need to add this custom properties logic during triggering time also? if needed please let me know, I can update PR, already I have code but reverted.

### Why was this change needed?
Closes #2393

### CODE REVIEW

File: apps/api/src/app/subscribers/usecases/update-subscriber/update-subscriber.command.ts [hunk 1]
```diff
@@ -28,6 +28,9 @@ export class UpdateSubscriberCommand extends EnvironmentCommand {
   @IsOptional()
   locale?: string;
 
+  @IsOptional()
+  data?: { [key: string]: string };
```
[the remark this answers is not in the document]
Comment (tsssdev) [reply apps/api/src/app/subscribers/usecases/update-subscriber/update-subscriber.command.ts:32]: > @tsssdev do you think maybe we should match this to the interface of the @novu/node package to allow boolean and numbers?

Yes @scopsy , will apply the changes

[the remark this answers is not in the document]
Comment (tsssdev) [reply apps/api/src/app/subscribers/usecases/update-subscriber/update-subscriber.command.ts]: @scopsy created new type for subscriber data and used all places and also applied UI changes to display subscriber data in subscriber list page, can you please review again and let me know if any other changes needed?

**Review by scopsy — APPROVED** [review]
Love the shared CustomData type!

**Comment by scopsy** [comment]
could you please take a look at the codesee errors? https://deepsource.io/gh/novuhq/novu/run/9789332f-3c07-4f53-be44-823025fc4e42/javascript/JS-0455?listindex=0

**Comment by tsssdev** [comment]
> could you please take a look at the codesee errors? https://deepsource.io/gh/novuhq/novu/run/9789332f-3c07-4f53-be44-823025fc4e42/javascript/JS-0455?listindex=0

Sure @scopsy let me check

**Comment by tsssdev** [comment]
> > could you please take a look at the codesee errors? https://deepsource.io/gh/novuhq/novu/run/9789332f-3c07-4f53-be44-823025fc4e42/javascript/JS-0455?listindex=0
> 
> Sure @scopsy let me check

Done @scopsy now codesee check is successful

**Pull request merged**